specs/interop/predeploys.md

```diff
@@ -130,15 +231,23 @@ hash comparison.
 A simple implementation of the `validateMessage` function is included below.
 
 ```solidity
-    function validateMessage(Identifier calldata _id, bytes32 _msgHash) external {
-        // We need to know if this is being called on a depositTx
-        if (IL1BlockInterop(Predeploys.L1_BLOCK_ATTRIBUTES).isDeposit()) revert NoExecutingDeposits();
+function validateMessage(Identifier calldata _id, bytes32 _msgHash) external {
+  bytes32 checksum = calculateChecksum(_id, _msgHash);
 
-        emit ExecutingMessage(_msgHash, _id);
-    }
+  (bool _isSlotWarm,) = _isWarm(checksum);
+
+  if (!_isSlotWarm) revert NotWarm();
+
+  emit ExecutingMessage(_msgHash, _id);
 }
 ```
 
+`calculateChecksum` implements the checksum computation (including type-byte) as defined
+in the [access-list checksum computation](#type-3-checksum) spec.
+
+`_isWarm` checks that the access-list prepared the `checksum` storage key to be warm,
+no other contract function may warm up this storage without message validation.
```

Comment:
```suggestion
`_isWarm` checks that the access-list prepared the `checksum` storage key to be warm.
**No other contract function may warm up this storage without message validation.**
```
this is extremely critical. we should add the relevant in-line comments and nat-spec in `CrossL2Inbox.col`